### PR TITLE
GitHub branch and last-commit info

### PR DESCRIPTION
**Pull request by lowsky**
I needed this additionally information about the branches of a repo and it's last commit.
This can be used in the query like this:

```
  repo(ownerUsername: "clayallsopp", name: "graphqlhub") {
    branches(limit:2) {
      name
      last_commit {
        ... CommitInfoFragment
      }
    }
  }
```

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -157,3 +157,6 @@ sftp-config.json
 
 .env
 dist/
+
+## graphql - generated code
```
Comment (clayallsopp) [inline .gitignore:161]: any more context on what this is? I'm not seeing these locally - is it something your editor added?

File: graphqlhub-schemas/src/github.js [hunk 2]
```diff
@@ -187,6 +188,21 @@ let IssueType = new GraphQLObjectType({
 });
 
 
+let BranchType = new GraphQLObjectType({
+  name : 'GitBranch',
```
Comment (clayallsopp) [inline graphqlhub-schemas/src/github.js:192]: nit - can we make this `GithubBranch` to keep in line with the rest of the names?

**Comment by clayallsopp** [comment]
@lowsky thanks, this is great! makes a lot of sense - I'll make a few small changes to tidy things up, will tag you on that commit

File: .gitignore (hunk 1 above)
Comment (lowsky) [inline .gitignore:161]: These files were generated somehow while any build step, so I added this line, to not accidentally adding them.

File: graphqlhub-schemas/src/github.js (hunk 2 above)
Comment (lowsky) [inline graphqlhub-schemas/src/github.js:192]: Yea, makes fully sense!

**Comment by lowsky** [comment]
Thanks for merging!

**Pull request merged**